src/NServiceBus.RabbitMQ/RabbitMQTransportInfrastructure.cs

```diff
@@ -25,13 +25,24 @@ public RabbitMQTransportInfrastructure(SettingsHolder settings, string connectio
             var connectionConfiguration = new ConnectionStringParser(settings).Parse(connectionString);
             connectionFactory = new ConnectionFactory(connectionConfiguration);
 
+            SetUsePublisherConfirmsIfPresent(settings, connectionConfiguration);
+
             CreateTopology();
 
             channelProvider = new ChannelProvider(connectionFactory, routingTopology, connectionConfiguration.UsePublisherConfirms);
 
             RequireOutboxConsent = false;
         }
 
+        static void SetUsePublisherConfirmsIfPresent(SettingsHolder settings, ConnectionConfiguration connectionConfiguration)
+        {
+            if (settings.HasSetting(SettingsKeys.UsePublisherConfirms))
+            {
+                var publisherCofirms = settings.Get<bool>(SettingsKeys.UsePublisherConfirms);
+                connectionConfiguration.UsePublisherConfirms = publisherCofirms;
+            }
```

Comment:
Yeah, for the purposes of this PR, I think the code is good to go.

I do think that there is scope for some good refactoring here, either in core, the transport, or both.